### ISSUE TITLE
rg leads to 1000% cpu usage

### ISSUE THREAD

**Issue by yantze** [microsoft/vscode#124230]
Issue Type: <b>Bug</b>

step1: cmd+shift+p and stroke "open recent"
step2: select a big repo such as vscode source
step3: wait a moment and htop show this.

Here is screenshot:
![image](https://user-images.githubusercontent.com/2226423/118949712-18595600-b98c-11eb-8f9c-bd023b9192c5.png)

https://user-images.githubusercontent.com/2226423/118948843-42f6df00-b98b-11eb-8e13-cb7ed5d741d4.png
![image](https://user-images.githubusercontent.com/2226423/118948843-42f6df00-b98b-11eb-8e13-cb7ed5d741d4.png)

The complete command is as follows:
```
/Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar.unpacked/vscode-ripgrep/bin/rg --files --hidden --case-sensitive -g **/package.json -g !**/.git -g !**/.svn -g !**/.hg -g !**/CVS -g !**/.DS_Store -g !**/*.pyc -g !**/__pycache__ -g !**/.*.un~ -g !**/.classpath -g !**/.project -g !**/.settings -g !**/.factorypath --no-ignore --follow --no-config --no-ignore-global
```

VS Code version: Code 1.56.2 (054a9295330880ed74ceaedda236253b4f39a335, 2021-05-12T17:44:30.902Z)
OS version: Darwin x64 19.6.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|13, 11, 7|
|Memory (System)|16.00GB (0.32GB free)|
|Process Argv|--crash-reporter-id 52ec8d52-ccae-4fa5-a23c-2f7b6fccbee6|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (25)</summary>

Extension|Author (truncated)|Version
---|---|---
browse-lite|ant|0.2.8
codestream|Cod|11.0.7
vscode-eslint|dba|2.1.20
vscode-babel-coloring|dza|0.0.4
gitlens|eam|11.4.1
EditorConfig|Edi|0.16.4
vscode-firefox-debug|fir|2.9.4
code-runner|for|0.11.4
rest-client|hum|0.24.5
iceworks-refactor|ice|0.1.0
vscode-redux-devtools|jin|1.0.2
bash-ide-vscode|mad|1.11.0
vscode-language-pack-zh-hans|MS-|1.56.2
python|ms-|2021.5.842923320
vscode-pylance|ms-|2021.5.3
jupyter|ms-|2021.6.832593372
remote-ssh|ms-|0.65.4
remote-ssh-edit|ms-|0.65.6
debugger-for-chrome|msj|4.12.12
color-highlight|nau|2.3.0
code-spell-checker|str|1.10.2
vscode-status-bar-format-toggle|tom|2.0.0
vscodeintellicode|Vis|1.2.14
vim|vsc|1.20.3
markdown-all-in-one|yzh|3.4.0

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551:30291414
vspre833cf:30267465
pythonptprofiler:30281270
vshan820:30294714
vscorecescf:30290706
pythondataviewer:30285071
vscus158cf:30286554
vscgsv2:30301612
vscorehovct:30302760
bridgeflight:30302069
vscod805:30301674

```

</details>

**Comment by roblourens** [comment]
This is probably triggered by a python-related extension. You can get the logs following the steps here to figure out which one https://github.com/microsoft/vscode/wiki/Search-Issues

**Comment by yantze** [comment]
Disable all extensions and remove all my settings.json from "/Users/yantze/Library/Application Support/Code/User/"，CPU also 1000%.

I found a way to consistently reproduce：
1. add multiple folder to workspace
2. close others editor window
3. reload window again.

**Comment by roblourens** [comment]
It must be coming either from an extension or from you triggering a search with `cmd+p`. If you get the logs there should be more clues.

**Comment by ayqy** [comment]
The same problem is here

![image](https://user-images.githubusercontent.com/12402989/119220325-8fbdef80-bb1c-11eb-8db7-93a4d8847934.png)

It's still here after all extensions disabled.

**Comment by ayqy** [comment]
![image](https://user-images.githubusercontent.com/12402989/119220382-d0b60400-bb1c-11eb-95c9-8c86189523ac.png)
It seems the problem occurred from 1.56, everything is fine before this main update, and the tow patch have no help.

Version: 1.56.2
Commit: 054a9295330880ed74ceaedda236253b4f39a335
Date: 2021-05-12T17:44:30.902Z (1 wk ago)
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Darwin x64 19.6.0

I can reproduce definitely 100%, fell free to ping me for any other detail debug logs.

**Comment by ayqy** [comment]
Reproduce steps:
1.Add three-four big repos to workspace with many many node_modules (which is my daily work);
2.CMD + Q force quit vscode;
3.Just click Dock icon to start vscode, and then my machine comes to be crazy...CPU BOOM!!!
---
4.Ten minutes later, one rg process died, the others are still crazy;
5.And then simply kill them, everything including search seems normally, no related error found from DevTools Console;
6.So, I am afraid of restart my vscde now 😂

**Comment by jltml** [comment]
Looks like I'm having this issue as well — I'm pretty unfamiliar with VS Code, but I ran into a possibly similar issue with Atom a couple months ago. It turned out to be that its fuzzy-finder package was using a lot of the CPU when the user folder was opened — it was trying to index every single folder and file in my user directory. When the multiple instances of `rg` were just using up my CPU, I was also editing files within my home directory — so manybe something like that is going on? Again, though, I'm not much of a VS Code person, so I could totally be wrong. (I can't seem to find the exact issue I'm thinking of in the Atom repo, but if I find it later, I'll update this.)

**Comment by roblourens** [comment]
Can someone get the logs that I asked for above?

**Comment by yantze** [comment]
> It must be coming either from an extension or from you triggering a search with `cmd+p`. If you get the logs there should be more clues.

Where are the logs? `~/Library.../Code/logs` ? or Output Panel ?

**Comment by yantze** [comment]
No need `cmd+p`，it just open the workspace with project.code-workspace and rg is 1000%

**Comment by yantze** [comment]
I found a resolution: Close "Follow Symlinks"
![image](https://user-images.githubusercontent.com/2226423/120263885-30f03700-c2cf-11eb-87a7-91549228ac4e.png)


Because some special project the node_modules has many symlinks.
```
project
 - node_modules
     - package-a (symlink _package-a)
     - package-b (symlink _package-b)
     - _package-a@1.2.3
     - _package-b@1.2.3
```

**Comment by yantze** [comment]
Maybe it would be better to ignore node_modules based on the search settings("Search: Exclude"):
![image](https://user-images.githubusercontent.com/2226423/120264160-cb507a80-c2cf-11eb-98d2-0017d504d7b2.png)

**Comment by roblourens** [comment]
Glad you found that option. Likely some extension triggered a search without excluding node_modules. The steps to see the logs are here https://github.com/microsoft/vscode/wiki/Search-Issues#filing-a-search-issue

**Comment by ayqy** [comment]
Version: 1.57.1
Commit: 507ce72a4466fbb27b715c3722558bb15afa9f48
Date: 2021-06-17T13:28:32.912Z
Electron: 12.0.7
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Darwin x64 19.6.0

1.57.1 fixed my problem, CPU boom from 1.56 - 1.57
